### DISCUSSION TITLE
Is there a way to make custom pose for openpose preprocessor?

### DISCUSSION THREAD

**Discussion by Drakmour** [Mikubill/sd-webui-controlnet#161] [General]
Is there a way to make custom pose for openpose preprocessor? Something like a posemaker like this one - https://webapp.magicposer.com/ with a possibility to extract the pose with the "skeleton" similar to what extension makes with openpose and feed it to an extension?

**Comment by ljleb** [comment]
There's this blender scene also if you want to make your own poses. It needs a bit of tweaking if you don't own the autorig plugin, but otherwise worked perfectly for me: https://toyxyz.gumroad.com/l/ciojz

Suggested tweaks: add camera, lock camera to view. F12 to take a photo, change resolution in scene settings to 512x512 or matching output resolution.

**Comment by Drakmour** [comment]
Oh, so you just use this skeleton as initial image and the extension reads this skeleton to build own skeleton? Neat. Btw is there a way to determine whether character stays front or backward? Those horns on the head should point forward to be detected as back to viewer? And is there a way to show that limb or part body are behind something with this pose skeleton? Maybe make a special coloration, that turns brighter \ different color if it is closer\further to viewer or if it is behind something. Or until we get skeleton + depth it would be not impossible?

**Comment by Drakmour** [comment]
Oh and is it possible to make OpenPose hand? The one that also makes fingers! That would be fantastic! Downloading Blender right now. :-)

**Comment by ljleb** [comment]
The "horns" orientation and shape definitely influence the result a lot. I think I read somewhere the head vertices correspond to ears and eyes. The colors may or may not influence the orientation as well, I need to test it a bit more. They always seem to have the same pattern in openpose data.

I tried putting the openpose stickman as far away as possible just to see what it would do. It seems that without a very suggestive prompt, the sampler stops following guidance from controlnet openpose model when the stickman is too far away.

![tmph8zvakn3-1](https://user-images.githubusercontent.com/32277961/219692334-330b80f7-5ec7-44c4-a190-34353cd5e13c.png)

![24987-3901125937-masterpiece, best quality, highres,_view from above, distant view, extremely far away, strategic camera, high altitude view_BREA](https://user-images.githubusercontent.com/32277961/219691976-cbb12c62-625b-4d57-9536-3c343779ae18.png)


As far as my testing goes, it does not seem the openpose control model was trained with hands in the dataset. Would love to see a controlnet capable of honouring hand openpose data!

**Comment by Drakmour** [comment]
There is also an interesting idea in reddit. :-)
https://www.reddit.com/r/StableDiffusion/comments/114jddr/using_controlnet_with_a_posable_art_figurine/

**Reply by patrickgalbraith** [reply]
That one was me! Glad you liked it

**Comment by aoirusann** [comment]
A friendly reminder: these kinds of topics are usually done in "Discussions"(if exists), which can ease the burden on maintainers. 😸 ❤️

**Comment by Drakmour** [comment]
> A friendly reminder: these kinds of topics are usually done in "Discussions"(if exists), which can ease the burden on maintainers. 😸 ❤️

Oh, sorry! It's just some extensions don't have a discussion tab and I had to make threads in Issues, got a bit confused after that and automatically made here too. :-)

**Comment by ljleb** [comment]
There should be a button on the right "convert to discussion", I believe only users with the maintainer or developer role have access to it though.

**Comment by Mikubill** [comment]
Agree and converted.

**Comment by Drakmour** [comment]
@ljleb @Mikubill Thank you!

**Comment by matrix4767** [comment]
One extension is in the works, so hopefully it will be done soon.

**Reply by ljleb** [reply]
do you have a link for it?

**Comment by FugueSegue** [comment]
It would be nice to be able to edit the skeleton. Or at least provide some sort of skeleton image of my own.

This extension does a pretty good job of interpreting a photo and guessing the skeleton joint positions. But it isn't always correct. It would be great if I was able to modify the generated skeleton in some sort of 2d editor within the webui. But that would increase the complexity of the extension.

If I could just upload a simple PNG file with 9 pixel squares like this and have the program interpret it as a skeleton:

![tmpxdr4vue_demo](https://user-images.githubusercontent.com/59415458/219868233-7f4eaa3d-d46f-4922-83e0-558538f21b90.png)

**Reply by Drakmour** [reply]
And we need to solve the problem with multiple charavters that overlap each other, it mutilates the skeleton cuz it grabs some bones from one, some from other. Tried to make 2 separate skeletons and then overlapped them myself and it was totally confused where is what.

**Comment by SepticSpoons** [comment]
Something like this? https://github.com/fkunn1326/openpose-editor

It's in Japanese, but supports English also. Adds an OpenPose Editor tab to webui and allows you to add skeletons, move parts, detect movement from an image+edit it on the fly, and then send it to controlnet. Make sure you set preprocessor to none and model:openpose

**Reply by FugueSegue** [reply]
This looks nice! I haven't tried it yet. So I don't know how well it works. I can't speak for the developer but IMHO this should be incorporated into the ControlNet extension itself.

**Reply by SepticSpoons** [reply]
There is a short preview video here.

<blockquote class="twitter-tweet"><p lang="ja" dir="ltr">OpenPose Editor for Automatic1111 webuiを公開しました！<br>webuiの拡張機能として動作するので、そのままControlnetの拡張機能に送ることができます！<br>ぜひ使ってみてください！<a href="https://t.co/dUHh7nmkfe">https://t.co/dUHh7nmkfe</a> <a href="https://t.co/MBYxvJKYEw">pic.twitter.com/MBYxvJKYEw</a></p>&mdash; どこかのFくん (@fkunn1326) <a href="https://twitter.com/fkunn1326/status/1627173198751211520?ref_src=twsrc%5Etfw">February 19, 2023</a></blockquote>

**Reply by jmilez** [reply]
If you get an error message when feeding a premade pose to controlnet, check for updates on the extension tab....
